qubes/vm/qubesvm.py

```diff
@@ -406,13 +406,29 @@ class QubesVM(qubes.vm.mix.net.NetVMMixin, qubes.vm.LocalVM):
             :param subject: Event emitter (the qube object)
             :param event: Event name (``'domain-start-failed'``)
 
+        .. event:: domain-pre-paused (subject, event)
+
+            Fired at the beginning of :py:meth:`paused` method and before
+            ``libvirt_domain.suspend()``.
+
```

Comment:
Since all places where this (and the next) event is fired is aN async function, better allow async handlers too.